bundles/org.openhab.binding.bluetooth.bluez/src/main/java/org/openhab/binding/bluetooth/bluez/internal/BlueZBluetoothDevice.java

```diff
@@ -131,14 +131,14 @@ public void dispose() {
             try {
                 dev.getAdapter().removeDevice(dev.getRawDevice());
             } catch (DBusException ex) {
-                if (ex.getMessage().contains("Does Not Exist")) {
-                    // this happens when the underlying device has already been removed
-                    // but we don't have a way to check if that is the case beforehand so
-                    // we will just eat the error here.
-                } else {
+                String exceptionMessage = ex.getMessage();
+                if (exceptionMessage == null || exceptionMessage.contains("Does Not Exist")) {
```

Comment:
Are you sure you do not want to rather check that exceptionMessage is not null?

Reply:
Yes, because it falls thru immediate and prevents the second part `equals` to throw an NPE. 
If you remove it, the compiler will throw a warning. And the debug format  method handles the null fine.